### PR TITLE
Fixing some minor bugs

### PR DESCRIPTION
**Pull request by jdalrymple**
- Updated from tslint to eslint since tslint will be deprecated soon
- Fixed the compilation of the browser file and cleaned up the rollup config
- Removed default exports
- Fixed some typing errors due to missing pointer to the index.d.ts file
- Added docs for browser usage
- Removed circular dependencies 

closes: #485 
closes: #475

### CODE REVIEW

File: test/integration/services/Labels.ts [hunk 1]
```diff
@@ -30,16 +30,17 @@ describe('Labels.create', () => {
...
+    expect(value).toEqual('');
   });
 });
 
 describe('Labels.all', () => {
   beforeAll(async () => {
     const labels: object[] = [];
 
-    for (let i = 0; i < 50; i++) {
+    for (let i = 0; i < 50; i += 1) {
```
Comment (casz) [inline test/integration/services/Labels.ts]: but why?

File: package.json [hunk 2]
```diff
@@ -88,18 +88,18 @@
     "url": "https://github.com/jdalrymple/node-gitlab"
   },
   "scripts": {
-    "build": "npm run build:cli && tsc && rollup -c",
+    "build": "npm run build:cli && tsc && rollup -c && mv dist/src dist/types",
```
[the remark this answers is not in the document]
Comment (jdalrymple) [reply package.json]: When looking at the published files, it seemed clearer to have all the d.ts files in a types folder then nested in /src

File: src/core/infrastructure/RequestHelper.ts [hunk 3]
```diff
@@ -42,7 +64,7 @@ export async function get(
   if (!query.page && underLimit && pagination.next) {
     const { next } = Li.parse(headers.link);
     const leaf = service.url.split('/').pop() || '';
-    const regex = new RegExp(`.+\/api\/v\\d(\/${leaf})?\/`);
+    const regex = new RegExp(`.+/api/v\\d(/${leaf})?/`);
```
[the remark this answers is not in the document]
Comment (jdalrymple) [reply src/core/infrastructure/RequestHelper.ts]: It was suggested by eslint ? :s, but yea will do

File: test/integration/services/Labels.ts (hunk 1 above)
Comment (jdalrymple) [reply test/integration/services/Labels.ts]: eslint

**Pull request merged**